presto-orc/src/main/java/com/facebook/presto/orc/OrcWriterOptions.java

```diff
@@ -316,4 +151,184 @@ public String toString()
+        {
+            this.stripeMaxSize = requireNonNull(stripeMaxSize, "stripeMaxSize is null");
+            return this;
+        }
+
+        public Builder withStripeMaxRowCount(int stripeMaxRowCount)
+        {
+            checkArgument(stripeMaxRowCount >= 1, "stripeMaxRowCount must be at least 1");
+            this.stripeMaxRowCount = stripeMaxRowCount;
+            return this;
+        }
+
+        public Builder withRowGroupMaxRowCount(int rowGroupMaxRowCount)
+        {
+            checkArgument(rowGroupMaxRowCount >= 1, "rowGroupMaxRowCount must be at least 1");
+            this.rowGroupMaxRowCount = rowGroupMaxRowCount;
+            return this;
+        }
+
+        public Builder withDictionaryMaxMemory(DataSize dictionaryMaxMemory)
+        {
+            this.dictionaryMaxMemory = requireNonNull(dictionaryMaxMemory, "dictionaryMaxMemory is null");
+            return this;
+        }
+
+        public Builder withMaxStringStatisticsLimit(DataSize maxStringStatisticsLimit)
+        {
+            this.maxStringStatisticsLimit = requireNonNull(maxStringStatisticsLimit, "maxStringStatisticsLimit is null");
+            return this;
+        }
+
+        public Builder withMaxCompressionBufferSize(DataSize maxCompressionBufferSize)
+        {
+            this.maxCompressionBufferSize = requireNonNull(maxCompressionBufferSize, "maxCompressionBufferSize is null");
+            return this;
+        }
+
+        public Builder withCompressionLevel(OptionalInt compressionLevel)
+        {
+            this.compressionLevel = requireNonNull(compressionLevel, "compressionLevel is null");
+            return this;
+        }
+
+        public Builder withStreamLayout(StreamLayout streamLayout)
+        {
+            this.streamLayout = requireNonNull(streamLayout, "streamLayout is null");
+            return this;
+        }
+
+        public Builder withIntegerDictionaryEncodingEnabled(boolean integerDictionaryEncodingEnabled)
+        {
+            this.integerDictionaryEncodingEnabled = integerDictionaryEncodingEnabled;
+            return this;
+        }
+
+        public DataSize getStripeMinSize()
+        {
+            return stripeMinSize;
+        }
+
+        public DataSize getStripeMaxSize()
+        {
+            return stripeMaxSize;
+        }
+
+        public int getStripeMaxRowCount()
+        {
+            return stripeMaxRowCount;
+        }
+
+        public int getRowGroupMaxRowCount()
+        {
+            return rowGroupMaxRowCount;
+        }
+
+        public DataSize getDictionaryMaxMemory()
+        {
+            return dictionaryMaxMemory;
+        }
+
+        public DataSize getMaxStringStatisticsLimit()
+        {
+            return maxStringStatisticsLimit;
+        }
+
+        public DataSize getMaxCompressionBufferSize()
+        {
+            return maxCompressionBufferSize;
+        }
+
+        public OrcWriterOptions.Builder copy()
```

Comment:
I don't have a strong opinion against it, but it feels like a better interface could be a `OrcWriterOptions#builder(OrcWriterOptions options)` that creates a builder from an existing immutable instance of the `OrcWriterOptions`

Reply:
this is removed.